src/server/pps/server/api_server.go

```diff
@@ -458,32 +526,106 @@ func (a *apiServer) Version(version int64) error {
 	return nil
 }
 
-func (a *apiServer) AddShard(shard uint64, version int64) error {
-	pipelineInfos, err := a.ListPipeline(context.Background(), &ppsclient.ListPipelineRequest{})
+func (a *apiServer) AddShard(shard uint64) error {
```

Comment:
So I think there's a couple of issues with this implementation of `AddShard`.
If a pipeline gets created for shard `0` before `AddShard(0)` is called then you'll ignore the pipeline (because the channel for shard `0` will be nil) and you'll never see it again in the call to `SubscribePipelineInfos`. So we'll have a pipeline that isn't being run anywhere.